### ISSUE TITLE
RUSE — The journey so far: from project birth to RN-20M

### ISSUE THREAD

**Issue by bluegrouse92-lgtm** [N-A-pixel/Ruse-Public#5]
# RUSE — The journey so far

This issue records the early experimental history that led from initial local model/runtime work to the first Ruse-native RN-20M lineage.

The distinction between **project birth** and **GitHub repository birth** matters. Ruse existed before the repository was created; the project estimate is that the experiment itself is roughly **12–24 hours older than the repository**. GitHub timestamps are used where they are authoritative, while the pre-repository period is treated as project history.

## Before the repository

The early work explored a local-first developmental model programme in which model execution, training, provenance and evaluation could be controlled on owned hardware rather than treated as opaque external services.

The project was being developed primarily from a **Steam Deck**, which became both the development workstation and the main experimental laboratory.

## 20 September 2026 — the GitHub repository is created

The `N-A-pixel/RUSE` repository was created at **2026-09-20 19:14:58 UTC**.

Commit `6405dcc` — **Establish Ruse application and training foundation** — landed at **19:18:52 UTC**. The historical commit title is retained here because it is part of the repository record; this public mirror does not document private product/application design.

## First experimental direction — existing open-weight models

The original learning route used existing open-weight models as experimental students and teacher candidates.

The project verified a real ARM64 runtime path including Python corpus/training tooling, Hugging Face checkpoints, HF → GGUF conversion, GGUF quantization and native `llama.cpp` execution.

A deterministic **108-example authored corpus** and validation/rejection infrastructure were established. Runtime success was deliberately separated from semantic competence: successfully loading and executing a model did not count as evidence that the model had learned anything useful.

The first frozen student baseline used **Qwen3-0.6B**:

```text
PASS:   0
FAIL:   6
UNSAFE: 0
```

Teacher-candidate probing also tested Llama/Phi-family artifacts. Those experiments exposed unsafe answers, weak task performance and teacher-identity contamination, which strengthened the validation/rejection paths.

## The Steam Deck changes the direction

A Qwen3-0.6B rank-8 LoRA experiment demonstrated that the Deck could load the BF16 base and attach the adapter, but the attempted optimizer step encountered unsafe memory pressure in a machine state where swap was already effectively exhausted.

That negative result sharpened the next research question:

> **If adapting a 0.6B pretrained model is already awkward on owned hardware, how small can a genuinely native model be while still supporting controlled learning research?**

# 21 September 2026 — RN-20M is created

RN-20M became the first native neural-model lineage initialized from deterministic random weights rather than inherited pretrained model weights.

```text
layers:                 10
hidden size:            384
FFN/intermediate:       1024
attention heads:        6
KV heads:               2
context:                1024
vocabulary:             16,000
parameters:             21,880,704   # historical early count
seed:                   20260921
pretrained weights:     NONE
```

A later preserved-artifact audit established the authoritative exact OG RN-20M parameter count as **21,865,344**; the earlier figure remains part of the historical record.

## Stage 0 — BIRTH

The first milestone asked only whether the model could be reproducibly built and executed.

```text
architecture
→ deterministic random initialization
→ project-generated tokenizer
→ Hugging Face checkpoint
→ HF → GGUF conversion
→ F16 GGUF
→ native llama.cpp
→ generated tokens
→ clean exit
```

The output was untrained nonsense, exactly as expected. Stage 0 proved deployment and reproducibility, not intelligence.

**Stage 0 verdict: PASS.**

## Stage 1 — controlled learnability

A synthetic codebook curriculum contained **176 training examples / 10,336 tokens**, while evaluation included held-out compositions and controls.

```text
training loss:             9.8869 → 1.6935
Day Zero eval loss:        9.8472
trained eval loss:         1.7386
held-out composition loss: 9.8677 → 1.7333
training wall time:         30.95 s
peak RSS:                   1.14 GB
```

The trained checkpoint was independently reloaded and reproduced the evaluation result.

That earned one narrow claim:

> **RN-20M can learn.**

Greedy exact generation remained **0/31**.

**Stage 1 verdict: PASS.**

## Stage B2 — generation improves, composition still fails

B2 corrected the answer-emission format and froze pass criteria before training, including a requirement to solve at least **4/6 genuinely unseen compositions**.

```text
Stage 1 exact generation: 0/31
B2 exact generation:      19/31
B2 token accuracy:        92.5%
seen colors:              6/6
known unknowns:           3/3
held-out compositions:    0/6
```

B2 therefore remained an unambiguous **FAIL** under its predeclared criteria.

The failure was still informative: the model moved from no exact answers to 19/31, and failed held-out outputs became malformed mixtures of learned components rather than random output.

**Stage B2 verdict: FAIL — with measurable new capability.**

## What happened in roughly a day

In a very compressed period, the experiment moved through:

```text
local model/runtime exploration
        ↓
ARM64 verification
        ↓
open-weight student + teacher experiments
        ↓
validation / rejection / provenance infrastructure
        ↓
Steam Deck training feasibility work
        ↓
questioning dependence on inherited pretrained weights
        ↓
RN-20M architecture
        ↓
Day Zero reproducible random model
        ↓
HF → GGUF → llama.cpp runtime
        ↓
Stage 1 controlled learning PASS
        ↓
B2 exact-generation experiment
        ↓
first informative native developmental failure
```

## Methodological pattern

The project repeatedly refused to collapse different claims into one another:

- a model running does not mean it learned;
- loss decreasing does not mean it can generate correctly;
- memorizing seen answers does not mean it can compose unseen answers;
- teacher output is not automatically trusted training data;
- a failed experiment is not discarded when it reveals a capability boundary.

## Lineage state at this point

```text
Real model/runtime pipeline                                  ✅
Native model from deterministic random initialization       ✅
Native GGUF / llama.cpp execution                            ✅
Measurable learning from scratch                             ✅
Learned state survives independent checkpoint reload        ✅
Exact generation of many learned answers                    ✅
Reliable unseen compositional generation                    ❌
General language competence                                 NOT CLAIMED
```

The rule going forward remained: **freeze the question before the experiment, preserve failures, and only claim what the evidence earns.**

The key result was the creation of a native model with a known initialization date, known ancestry, known training history, known failures and no inherited pretrained weights.

**Comment by bluegrouse92-lgtm** [comment]
# 23 September 2026 — the first E3 cohort graduates

The project history has reached another clean boundary.

The V1.1 controlled scaling family — RN-20M Clone (21,880,704 parameters), corrected RN-50M (50,046,720) and RN-100M (99,771,392) — completed the same frozen E3 curriculum from exactly 2,000,000 to **5,000,064 cumulative tokens**.

Each model consumed **3,000,064 E3 tokens (23,438 × 128)** from stream:

`61b26ba926aa511aa777af8f23a11e304fe5dd7cd3f1ebb0589cf342dd329138`

All three wrote both recovery checkpoints, completed frozen validation/holdout evaluation and diagnostic continuations, and produced canonical final checkpoints.

Measured training on one Steam Deck:

- RN-20M Clone: **1.47 h / 566 tok/s / 962 MB peak RAM**
- RN-50M corrected: **3.24 h / 257 tok/s / 1,382 MB**
- RN-100M: **7.13 h / 117 tok/s / 2,223 MB**

Total: **9,000,192 additional model-training tokens in 11.84 hours (~11 h 50 m) aggregate raw training time.**

Post-E3 holdout losses were **4.1961 / 4.1584 / 4.1703** respectively. Diagnostic generation became more language-shaped but remained semantically weak, repetitive and attractor-prone; no conversational or intelligence claim is inferred from surface fluency.

Gary's separate 32K multilingual lineage had already reached E3 independently, so the first four-model developmental cohort is canonically frozen at the E3 rung.

The next research phase expands the family with the protected OG RN-20M, a four-model ~8M cohort, another ~100M model and a ~300M Oracle/Ruse School-scale lineage, then evaluates how heterogeneous model capabilities can be coordinated under the Legion research architecture.

**Comment by bluegrouse92-lgtm** [comment]
## E3-EI — targeted English intervention complete

A small post-E3 English intervention was run across RN-20M-clone, RN-50M-corrected, RN-100M and Gary. The three V1.1 models consumed 100,096 tokens each; Gary consumed 100,736 through his separate 32K tokenizer. All resumed their canonical E3 optimizer states.

The intervention produced a strong behavioral shift but did not reliably correct the targeted weaknesses. Historical validation and holdout loss worsened for all four, and older attractors were largely replaced by new narrative/literary attractors rather than removed.

The result is retained as a curriculum-design finding: a relatively small concentrated intervention can strongly redirect generation behavior at this developmental age without producing broad semantic-control improvement.

**E3-EI is frozen as an experimental descendant branch and is not promoted into the main developmental lineage.** Canonical E3 checkpoints remain immutable.
